### ISSUE TITLE
Improve support for code blocks in included markdown files

### ISSUE THREAD

**Issue by AdarshRawat1** [doxygen/doxygen#11225]
## Describe the bug
When a markdown is included in other markdown using `\include{doc} "./Secondary.md"` Then, the output of code snippets breaks. 

## Screenshots

- ![image](https://github.com/user-attachments/assets/cbdea7e1-c561-4e5f-ae2e-70eb283d6775)

- ![image](https://github.com/user-attachments/assets/b9bbb603-3794-40a9-b793-f69f2061fd60)

## Observations 
This happens only if there is formatting in the primary markdown - like ( **BOLD** , _Italic_ ). **if we remove the formattings from primary markdown then everything works fine**, but we can't avoid using formatting, so I tried few workarounds one of them worked. 

i.e. There is a need to handle the formatting edge cases.

## Temporary Fix
- This version gives correct output but `\code{.c}` doesn't work on GitHub Renders, so I have to keep 2 copies of the same code in docs. (One for Doxygen and one for GitHub )
``` 
\code{.c}
    struct secondary_struct {
        int a ; 
        int b ; 
    }
\endcode

```

p.s. I've been on this for quite some time 

## To Reproduce
Use this Zip -  [includedocs.zip](https://github.com/user-attachments/files/17690916/includedocs.zip)

## Expected behavior
Correctly formatted code snippets

## Version
1.12.0 (Windows 11, WSL, Ubuntu) 

## Additional context
- [Related Stack overflow issue ](https://stackoverflow.com/questions/52120027/doxygen-how-to-include-a-markdown-page-to-document-a-group)

- https://github.com/doxygen/doxygen/issues/7430

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11226

**Comment by albert-github** [comment]
The problem with the undesired `#!/bin/bash` in the code section cannot be solved directly by doxygen as shell scripts are not supported by doxygen.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by AdarshRawat1** [comment]
> The problem with the undesired `#!/bin/bash` in the code section cannot be solved directly by doxygen as shell scripts are not supported by doxygen.

When the bash script is in a normal markdown( primary markdown) then it works, but breaks only for included codes i.e. `\include{doc} "./Secondary.md"`

**Comment by albert-github** [comment]
@AdarshRawat1 

Can you please give an example so we are sure we use the same example / see the same problem:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

and report the warning / an image indicating what is wrong.

**Comment by AdarshRawat1** [comment]
@albert-github 

I had attached the zip in issue description for the initial issue. For bash specifically, kindly use this file - [bash.zip](https://github.com/user-attachments/files/17803699/bash.zip)

## Output when the bash script is in the primary markdown file 
- ![image](https://github.com/user-attachments/assets/fc61f4c0-8fad-49b3-9173-c5e5a5283842)

## Output when we use `includedocs` to add a bash script from a secondary markdown file 

- ![image](https://github.com/user-attachments/assets/e0e98414-7389-41b2-947d-67e35d5c3fb3)

## Temporary fix using `\code{}` 

- ![image](https://github.com/user-attachments/assets/a53f040d-b4a5-4dbc-8eb2-1bf73a604599)

**Comment by AdarshRawat1** [comment]
Thank you @albert-github this looks great.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
